lib/navigation/service/app_router_delegates.dart

```diff
@@ -0,0 +1,182 @@
+        final key = (route.settings as MaterialPage).key.toString();
+        final pathFromKey = key.substring(3, key.length - 3);
+
+        final uri = Uri.parse(pathFromKey);
+        final routePath = PathFactory.from(uri);
+
+        if (!(routePath is UnknownPath)) {
+          _navigationState.path = routePath;
+        }
+
+        return true;
+      },
+    );
+  }
+
+  @override
+  RoutePath get currentConfiguration {
+    print('\n---------------\n'
+        'getConfiguration: $_navigationState'
+        '\n-------------');
+
+    if (!_navigationState.isInitialized) {
+      return RootPath();
+    }
+
+    // if (!_authProvider.isAuthenticated) {
+    //   return LoginPath();
+    // }
+
+    return _navigationState.path;
+  }
+
+  @override
+  Future<void> setNewRoutePath(RoutePath configuration) async {
+    print('\n---------------\n'
+        'setNewRoute: $configuration'
+        '\n$_navigationState'
+        '\n-------------');
+
+    _navigationState.reset();
+
+    if (configuration is LoginPath) {
+      // await _authProvider.signOut();
+      if (_authProvider.isAuthenticated) {
+        _navigationState.path = HomePath();
+        return;
+      } else {
+        _navigationState.isDrawerExtended = false;
+      }
+    }
+
+    if (configuration is HomePath) {
+      _navigationState.selectedTab = 0;
+    } else if (configuration is TimetablePath) {
+      _navigationState.selectedTab = 1;
+    } else if (configuration is PortalPath) {
+      _navigationState.selectedTab = 2;
+    } else if (configuration is PeoplePath) {
+      _navigationState.selectedTab = 3;
+    }
+
+    _navigationState.path = configuration;
+  }
+}
+
+class InnerRouterDelegate extends AppRouterDelegate {
+  InnerRouterDelegate({@required NavigationStateProvider navigationState})
+      : _navigationState = navigationState;
+
+  final GlobalKey<NavigatorState> _navigatorKey = GlobalKey<NavigatorState>();
+  final NavigationStateProvider _navigationState;
+
+  @override
+  GlobalKey<NavigatorState> get navigatorKey => _navigatorKey;
+
+  @override
+  Widget build(BuildContext context) {
+    return Navigator(
+      key: _navigatorKey,
+      pages: <Page<dynamic>>[
+        MaterialPage(
+          child: _navigationState.path.page,
+          key: ValueKey<String>(_navigationState.path.location),
+        ),
+        if (_navigationState.customView != null)
+          MaterialPage(
+            child: _navigationState.customView,
+            key: ValueKey<String>('${_navigationState.customView.hashCode}'),
+          )
+      ],
+      onPopPage: (route, result) {
+        if (!route.didPop(result)) {
+          return false;
+        }
+
+        _navigationState.reset();
+        notifyListeners();
+        return true;
+      },
+    );
+  }
+
+  @override
+  Future<void> setNewRoutePath(RoutePath configuration) async {
+    assert(false);
```

Comment:
What's the assertion for?

Reply:
The method had to be overriden. I replaced it with a return just to have it clear that this method should exist and do nothing.